homeassistant/components/nexia/climate.py

```diff
@@ -132,61 +146,57 @@ async def async_setup_entry(
     )
 
-    entities = []
+    entities: list[NexiaZone] = []
     for thermostat_id in nexia_home.get_thermostat_ids():
-        thermostat = nexia_home.get_thermostat_by_id(thermostat_id)
+        thermostat: NexiaThermostat = nexia_home.get_thermostat_by_id(thermostat_id)
         for zone_id in thermostat.get_zone_ids():
-            zone = thermostat.get_zone_by_id(zone_id)
+            zone: NexiaThermostatZone = thermostat.get_zone_by_id(zone_id)
             entities.append(NexiaZone(coordinator, zone))
 
-    async_add_entities(entities, True)
+    async_add_entities(entities)
 
 
 class NexiaZone(NexiaThermostatZoneEntity, ClimateEntity):
     """Provides Nexia Climate support."""
 
-    def __init__(self, coordinator, zone):
+    def __init__(
+        self, coordinator: NexiaDataUpdateCoordinator, zone: NexiaThermostatZone
+    ) -> None:
         """Initialize the thermostat."""
         super().__init__(
             coordinator, zone, name=zone.get_name(), unique_id=zone.zone_id
         )
-        self._undo_humidfy_dispatcher = None
-        self._undo_aircleaner_dispatcher = None
+        unit = self._thermostat.get_unit()
+        min_humidity, max_humidity = self._thermostat.get_humidity_setpoint_limits()
+        min_setpoint, max_setpoint = self._thermostat.get_setpoint_limits()
         # The has_* calls are stable for the life of the device
         # and do not do I/O
         self._has_relative_humidity = self._thermostat.has_relative_humidity()
         self._has_emergency_heat = self._thermostat.has_emergency_heat()
         self._has_humidify_support = self._thermostat.has_humidify_support()
         self._has_dehumidify_support = self._thermostat.has_dehumidify_support()
-
-    @property
-    def supported_features(self):
-        """Return the list of supported features."""
-        supported = (
-            SUPPORT_TARGET_TEMPERATURE_RANGE
-            | SUPPORT_TARGET_TEMPERATURE
-            | SUPPORT_FAN_MODE
-            | SUPPORT_PRESET_MODE
-        )
-
+        supported = NEXIA_SUPPORTED
         if self._has_humidify_support or self._has_dehumidify_support:
             supported |= SUPPORT_TARGET_HUMIDITY
-
         if self._has_emergency_heat:
             supported |= SUPPORT_AUX_HEAT
-
-        return supported
+        self._attr_supported_features = supported
+        self._attr_preset_modes = self._zone.get_presets()
+        self._attr_fan_modes = self._thermostat.get_fan_modes()
+        self._attr_hvac_modes = HVAC_MODES
+        self._attr_min_humidity = percent_conv(min_humidity)
+        self._attr_max_humidity = percent_conv(max_humidity)
+        self._attr_min_temp = min_setpoint
+        self._attr_max_temp = max_setpoint
+        self._attr_temperature_unit = TEMP_CELSIUS if unit == "C" else TEMP_FAHRENHEIT
+        self._attr_target_temperature_step = 0.5 if unit == "C" else 1.0
+        self._attr_preset_modes = self._zone.get_presets()
```

Comment:
Duplicate of line 184.

Reply:
Thanks https://github.com/home-assistant/core/pull/64220